### DISCUSSION TITLE
Nuxt dynamic subdomain routing

### DISCUSSION THREAD

**Discussion by k11q** [nuxt/nuxt#16382] [Questions]
Hi I am trying to configure a dynamic subdomain routing for my nuxt3 apps, for example, every user each get a custom subdomain.
E.g. user1.domain.com
user2.domain.com
user3.domain.com

what is the best possible way to do this in nuxt, wether to use middleware or configure the vue router.

What I hope to achieve is I can redirect the dynamic subdomain from my pages/users/[id] files, where the [id] becomes subdomain, and if theres a subpage under the id for example pages/users/[id]/[subpage] I want it to become user1.domain.com/subpages for example. Thank you for the help!

**Comment by TheAlexLichter** [comment]
Check out https://github.com/nuxt/framework/discussions/6057

**Reply by k11q** [reply]
Sorry I don't quite understand the referenced discussion. How do it fit my case?

**Reply by TheAlexLichter** [reply]
Because https://github.com/zernonia/keypress covers user-base subdomains too and is open source ☺️ 
Taking a look at this implementation should give you a bit of guidance 👍

**Comment by brettmcdade** [comment]
Did you find a solution? I'm searching for the same thing.
